### DISCUSSION TITLE
[Keycloak 23.0.1] Custom User Federation unable to integrate resteasy-jackson2-provider

### DISCUSSION THREAD

**Discussion by moritz31** [keycloak/keycloak#25265] [Q&A]
We are using keycloak since years and have a custom user federation which uses resteasy. 
On the latest keycloak version of 22.0.5 this is working without problems with Resteasy 6.2.4 and the following configuration
`        <dependency>
            <groupId>org.jboss.resteasy</groupId>
            <artifactId>resteasy-client</artifactId>
            <version>${resteasy.version}</version>
            <scope>provided</scope>
        </dependency>
        <dependency>
            <groupId>org.jboss.resteasy</groupId>
            <artifactId>resteasy-client-api</artifactId>
            <version>${resteasy.version}</version>
            <scope>provided</scope>
        </dependency>`
        
With keycloak 23 the federation breaks with the following issue:
` RESTEASY003145: Unable to find a MessageBodyReader of content-type application/json and type class`

I have looked into keycloak and it seems like before 23.0.0 resteasy-jackson2-provider was provided by keycloak itself and therefore it worked out of the box.
The logical next step for me was copy the dependency to keycloak, what i did was putting the dependency into the providers folder and adding the following to my pom.
`
        <dependency>
            <groupId>org.jboss.resteasy</groupId>
            <artifactId>resteasy-jackson2-provider</artifactId>
            <version>${resteasy.version}</version>
            <scope>provided</scope>
        </dependency>
`

With that i got an strange issue `Cannot invoke “org.jboss.jandex.ClassInfo.name()” because “currentClazz” is null`

I have now invested a lot of time and does not have a further clue on how to solve this. Can someone maybe help me ?

**Comment by Maximilian-Burow** [comment]
I'm having the same issue when trying to add resteasy-jackson2-provider as a dependency to a custom spi implementation. Should we file a bug report?

**Reply by xgp** [reply]
I have the same problem.

Issue opened here https://github.com/keycloak/keycloak/issues/25589
